homeassistant/components/zwave_js/humidifier.py

```diff
@@ -0,0 +1,225 @@
+        client: ZwaveClient,
+        info: ZwaveDiscoveryInfo,
+    ) -> None:
+        """Initialize thermostat."""
+        super().__init__(config_entry, client, info)
+
+        self._attr_supported_features = SUPPORT_MODES
+        self._attr_available_modes = []
+
+        self._current_mode = self.get_zwave_value(
+            HUMIDITY_CONTROL_MODE_PROPERTY,
+            command_class=CommandClass.HUMIDITY_CONTROL_MODE,
+        )
+
+        if self._current_mode:
+            if (
+                str(HumidityControlMode.HUMIDIFY.value)
+                in self._current_mode.metadata.states
+            ):
+                self._attr_device_class = HumidifierDeviceClass.HUMIDIFIER
+            else:
+                self._attr_device_class = HumidifierDeviceClass.DEHUMIDIFIER
+
+            for mode in self._current_mode.metadata.states:
+                self._attr_available_modes.append(
+                    self._current_mode.metadata.states[mode]
+                )
+
+        self._setpoint_values: dict[HumidityControlSetpointType, ZwaveValue] = {}
+
+        for setpoint_type in HumidityControlSetpointType:
+            setpoint_value = self.get_zwave_value(
+                HUMIDITY_CONTROL_SETPOINT_PROPERTY,
+                command_class=CommandClass.HUMIDITY_CONTROL_SETPOINT,
+                value_property_key=setpoint_type,
+                add_to_watched_value_ids=True,
+            )
+            if setpoint_value:
+                self._setpoint_values[setpoint_type] = setpoint_value
+
+    def _setpoint_type_for_current_mode(self) -> HumidityControlSetpointType:
+        if self._current_mode:
+            return HUMIDITY_CONTROL_MODE_SETPOINT_MAP.get(
+                int(self._current_mode.value), []
+            )
+        return None
+
+    @property
+    def _setpoint_value(self) -> ZwaveValue | None:
+        """Optionally return a ZwaveValue for a setpoint."""
+
+        setpoint_types = self._setpoint_type_for_current_mode()
+        if len(setpoint_types) == 0:
+            return None
+
+        setpoint_type = setpoint_types[0]
+        if (
+            setpoint_type is None
+            or setpoint_type not in self._setpoint_values
+            or (val := self._setpoint_values[setpoint_type]) is None
+        ):
+            raise ValueError("Value requested is not available")
+
+        return val
+
+    async def async_turn_on(self, **kwargs: Any) -> None:
+        """Set new target humidity."""
+        if self._current_mode:
+            if (
+                str(HumidityControlMode.AUTO.value)
+                in self._current_mode.metadata.states
+            ):
+                new_mode = HumidityControlMode.AUTO
+            elif (
+                str(HumidityControlMode.HUMIDIFY.value)
+                in self._current_mode.metadata.states
+            ):
+                new_mode = HumidityControlMode.HUMIDIFY
+            else:
+                new_mode = HumidityControlMode.DEHUMIDIFY
```

Comment:
Would it be better to keep track of the last mode that was active before turning off and go back to that mode when turning on? Is there a way to persist that information across reboot or is that not necessary?

Reply:
We should receive the current state from the driver when starting the integration. Is that not the case?

Reply:
I understand now. If the device is off, we don't know what mode it was before that.

Reply:
I don't think we need to make the last not off state persistent across restarts. We can't guarantee that the user didn't change modes during the time Home Assistant was shutdown. So loading a stored state from file may be incorrect for the user anyway.

It's ok to store a last non off state in memory of we think that's the best user experience.

Reply:
The device is "on" when the state is anything other than "off". Turning the device "off" involves updating the value of `self._current_mode`, and inversely when turning the device back on, we need to set the mode to something other than Off. My question is what should that new mode be (choices are: auto, humidify, de-humidify). Should I keep track of the previous mode and go back to it instead of the logic I implemented here? If the previous mode is stored in an instance variable on the entity, then if I were to set the device to "off" and restart home-assistant, which mode should we go back to then?

Reply:
>  If the device is off, we don't know what mode it was before that.

Yes this is correct, sorry github didn't show your comment while I was writing mine...

Reply:
After splitting this in two entities, there is no longer a "mode" exposed on the humidifier, only a on/off state, so this is no longer necessary